workflow-controller/controller-service/src/test/java/io/hyscale/controller/directive/PropsJsonHandlerTest.java

```diff
@@ -22,48 +22,47 @@
 import java.util.Map;
 import java.util.Map.Entry;
 
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
 
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 import io.hyscale.commons.exception.HyscaleException;
 import io.hyscale.commons.utils.WindowsUtil;
+import io.hyscale.controller.ControllerTestInitializer;
 import io.hyscale.controller.directive.impl.PropsJsonHandler;
 import io.hyscale.controller.util.ServiceSpecTestUtil;
 import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
 import io.hyscale.servicespec.commons.model.PropType;
 import io.hyscale.servicespec.commons.model.service.ServiceSpec;
 
+@SpringJUnitConfig(classes = ControllerTestInitializer.class)
 public class PropsJsonHandlerTest {
 
-    private static final String FILEPATH = "/servicespecs/props_update.hspec.yaml";
-
     private static final String JSON_PATH = HyscaleSpecFields.getPath(HyscaleSpecFields.props);
 
-    private static PropsJsonHandler propsHandler = new PropsJsonHandler();
+    @Autowired
+    private PropsJsonHandler propsHandler;
 
     private static ObjectNode serviceSpecNode = null;
 
     private static ServiceSpec oldServiceSpec = null;
 
     private static ServiceSpec updatedServiceSpec = null;
 
-    @BeforeAll
-    public static void beforeAll() {
+    @Test
+    public void updateProps() {
         try {
-            oldServiceSpec = ServiceSpecTestUtil.getServiceSpec(FILEPATH);
-            serviceSpecNode = (ObjectNode) ServiceSpecTestUtil.getServiceSpecJsonNode(FILEPATH);
+            oldServiceSpec = ServiceSpecTestUtil.getServiceSpec("/servicespecs/props_update.hspec.yaml");
+            serviceSpecNode = (ObjectNode) ServiceSpecTestUtil
```

Comment:
I think this initialisation can be moved to @BeforeAll annotation.
The update is the main test
Also do you see a commonality in these handler test cases ?